batch_test.go

```diff
@@ -49,7 +49,7 @@ func TestEventBatch_Clear(t *testing.T) {
 	ok, err := eb.Add(eventhub.NewEventFromString("Foo"))
 	assert.True(t, ok)
 	assert.NoError(t, err)
-	assert.Equal(t, 174, eb.Size())
+	assert.Equal(t, 163, eb.Size())
```

Comment:
Why did this change?

Reply:
Oh presumably due to fewer bytes in the payload.

Reply:
I'm not totally up to date on the actual encoding _but_ we are now properly able to omit the group sequence number when it's not filled out, since it's a pointer (vs the primitive).

Reply:
Okay, this is interesting.

So there's an optimization in the marshaller where, if there is a continuous run of nil fields, it can potentially omit those fields entirely. However, since we couldn't nil out the sequence number we'd always end up having to write a byte for each value (`TypeCodeNull`) for each field that was in between any set fields (ie, we can't leave gaps between filled in values).

Now, we don't have that issue anymore so, if we actually haven't filled in anything besides the message ID (which happens to be the first field) then the rest of the values can be omitted entirely. 

So what I'm trying to say is that we're doing our part to save people bytes.